app/assets/javascripts/discourse/app/components/modal/grant-badge.js

```diff
@@ -0,0 +1,80 @@
+import { action } from "@ember/object";
+import Component from "@ember/component";
+import Badge from "discourse/models/badge";
+import GrantBadgeController from "discourse/mixins/grant-badge-controller";
+import I18n from "I18n";
+import UserBadge from "discourse/models/user-badge";
+import { all } from "rsvp";
+import discourseComputed from "discourse-common/utils/decorators";
+import { extractError } from "discourse/lib/ajax-error";
+import getURL from "discourse-common/lib/get-url";
+
+export default class GrantBadgeModal extends Component.extend(
+  GrantBadgeController
+) {
+  loading = true;
+  saving = false;
+  selectedBadgeId = null;
+
+  init() {
+    this.set("loading", true);
+    all([
+      Badge.findAll(),
+      UserBadge.findByUsername(this.get("post.username")),
+    ]).then(([allBadges, userBadges]) => {
+      this.setProperties({
+        allBadges,
+        userBadges,
+        loading: false,
+      });
+    });
```

Comment:
This was shifted from the old `onShow` hook, would be good to know if this should be placed differently. I didn't quite see a relevant `{{ on...}}` modifier ( `onclick`?) so opted for placing in `init` instead.